src/components/cleaner.ts

```diff
@@ -55,16 +55,20 @@ export class Cleaner {
             .map(file => path.resolve(outdir, file))
         ).then(files => Promise.all(
             // Try to unlink the files, returning a Promise for every file
-            files.map(file => fs.promises.unlink(file).then(() => {
-                this.extension.logger.addLogMessage(`File cleaned: ${file}`)
-                // If unlinking fails, replace it with an rmdir Promise
-            }, () => fs.promises.rmdir(file).then(() => {
-                this.extension.logger.addLogMessage(`Folder removed: ${file}`)
-            }, () => {
-                this.extension.logger.addLogMessage(`Error removing file: ${file}`)
-            })))
+            files.map(file => {
+                const stats: fs.Stats = fs.statSync(file)
+                if (stats.isFile()) {
+                    fs.promises.unlink(file).then(() => {
+                        this.extension.logger.addLogMessage(`File cleaned: ${file}`)
+                    }, () => {
+                        this.extension.logger.addLogMessage(`Error removing file: ${file}`)
+                    })
+                } else {
+                    this.extension.logger.addLogMessage(`Not removing non-file: ${file}`)
+                }
+            })
```

Comment:
The current code is not correct. In the original code, `files.map(file => fs.promises.unlink(file).then(...))` returns `Promise<void>[]`. Then, `Promise.all(promises: Promise<void>[])` is called.

https://github.com/James-Yu/LaTeX-Workshop/blob/7467c884f116fffa11e2e60d427607dee46217ab/src/components/cleaner.ts#L56-L65

However, in the current code,  `files.map(file => {...})` returns `void[]`. 

I think we should rewrite the whole code of `cleanGlob`.